### DISCUSSION TITLE
Implement functionality to set multiple cells of a TileMap with one call

### DISCUSSION THREAD

**Discussion by Xkonti** [godotengine/godot-proposals#7698] [2D]
Currently the only options to set a multiple tiles are:
- Call `TileMap.set_cell` multiple tiles
- Use a pattern

In a popular use case of procedurally generating a game map, patterns are not very useful. The only tool left is the `set_cell`. Both GDScript and C# when calling that function need to go through a mapping layer that then calls the internal C++ implementation. It's often that hundreds (or thousands) of cells need to be set / changed per frame. Especially in case of dynamic loading/unloading of map segments.

I believe that adding an additional functions, something like `set_cells` and `set_area`:
- `set_cells` - allows for passing an array (list) of same data as is currently required by `set_cell`. This allows for sending a set of potentially not adjacent cells:
  ```cs
  var cellData = new CellData[]
  {
      new CellData { layer = 0, pos = tile1_pos, atlas = 0, atlasPos = tile1_atlasPos },
      new CellData { layer = 0, pos = tile2_pos, atlas = 0, atlasPos = tile2_atlasPos },
      new CellData { layer = 0, pos = tile3_pos, atlas = 0, atlasPos = tile3_atlasPos },
      new CellData { layer = 0, pos = tile4_pos, atlas = 0, atlasPos = tile4_atlasPos },
      ...
  };
  tileMap.SetCells(cellData);
  ```
- `set_area` - allows for passing data for updating a rectangular area (chunk) of the tile map. Data would take a form of a 2-dimensional array or something. This would offer the most room for optimizations.

would yield a number of benefits:
- Passing a list of cells to update at the same time equals to reducing the need for multiple calls between scripting runtime and the actual Godot implementation - better performance
- There would be multiple opportunities for additional optimizations as Godot's internal code can handle the larger chunk of data more optimally

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/1538.
